gcloud-java-dns/src/test/java/com/google/cloud/dns/DnsBatchTest.java

```diff
@@ -0,0 +1,632 @@
+    assertEquals(MAX_SIZE, size);
+    String selector = (String) capturedOptions.getValue().get(ZONE_LIST_OPTIONS[1].rpcOption());
+    assertEquals(PAGE_TOKEN, selector);
+    selector = (String) capturedOptions.getValue().get(ZONE_LIST_OPTIONS[2].rpcOption());
+    assertTrue(selector.contains(Dns.ZoneField.DESCRIPTION.selector()));
+    assertTrue(selector.contains(Dns.ZoneField.NAME.selector()));
+    selector = (String) capturedOptions.getValue().get(ZONE_LIST_OPTIONS[3].rpcOption());
+    assertEquals(DNS_NAME, selector);
+    // check the callback
+    ManagedZonesListResponse response = new ManagedZonesListResponse()
+        .setManagedZones(ZONE_LIST)
+        .setNextPageToken(PAGE_TOKEN);
+    RpcBatch.Callback<ManagedZonesListResponse> capturedCallback = callback.getValue();
+    EasyMock.verify(optionsMock);
+    EasyMock.reset(optionsMock);
+    EasyMock.expect(optionsMock.service()).andReturn(dns).times(ZONE_LIST.size());
+    EasyMock.replay(optionsMock);
+    capturedCallback.onSuccess(response);
+    Page<Zone> page = batchResult.get();
+    assertEquals(PAGE_TOKEN, page.nextPageCursor());
+    Iterator<Zone> iterator = page.values().iterator();
+    int resultSize = 0;
+    EasyMock.verify(dns);
+    EasyMock.reset(dns);
+    EasyMock.expect(dns.options()).andReturn(optionsMock).times(ZONE_LIST.size() + 1);
+    EasyMock.replay(dns);
+    Zone zoneInfoFunctional = new Zone(dns, new ZoneInfo.BuilderImpl(ZONE_INFO));
+    while (iterator.hasNext()) {
+      assertEquals(zoneInfoFunctional, iterator.next());
+      resultSize++;
+    }
+    assertEquals(ZONE_LIST.size(), resultSize);
+  }
+
+  @Test
+  public void testCreateZone() {
+    EasyMock.reset(batchMock);
+    Capture<RpcBatch.Callback<ManagedZone>> callback = Capture.newInstance();
+    Capture<Map<DnsRpc.Option, Object>> capturedOptions = Capture.newInstance();
+    Capture<ManagedZone> capturedZone = Capture.newInstance();
+    batchMock.addCreateZone(EasyMock.capture(capturedZone), EasyMock.capture(callback),
+        EasyMock.capture(capturedOptions));
+    EasyMock.replay(batchMock);
+    DnsBatchResult<Zone> batchResult = dnsBatch.createZone(ZONE_INFO);
+    assertEquals(0, capturedOptions.getValue().size());
+    assertEquals(ZONE_INFO.toPb(), capturedZone.getValue());
+    assertNotNull(callback.getValue());
+    try {
+      batchResult.get();
+      fail("No result available yet.");
+    } catch (IllegalStateException ex) {
+      // expected
+    }
+    // testing error here, success is tested with options
+    RpcBatch.Callback<ManagedZone> capturedCallback = callback.getValue();
+    capturedCallback.onFailure(GOOGLE_JSON_ERROR);
+    try {
+      batchResult.get();
+      fail("Should throw a DnsException on error.");
+    } catch (DnsException ex) {
+      // expected
+    }
+  }
+
+  @Test
+  public void testCreateZoneWithOptions() {
+    EasyMock.verify(dns);
```

Comment:
On the verify calls discussion (commenting here so it appears rather than getting compressed), I'm not quite convinced this is helpful because in `setUp` you create new mocks each time, so there isn't too much to verify in my mind.  I could misunderstand though; let me know if that's the case.